### DISCUSSION TITLE
Using Rich live full screen with pdb debugging

### DISCUSSION THREAD

**Discussion by miohtama** [Textualize/rich#2505] [General]
If I set a breakpoint using [ipdb](https://pypi.org/project/ipdb/) (or pdb to related matter), in Rich application that is in `rich.live.Live` mode, the console debugger does not work. I guess this is because Rich takes over terminal and pdb would need some sort of reset for the terminal to function correctly.

I am running my application as:

```python
Live(console=console, screen=True, auto_refresh=False)
```

And when it hits a breakpoint the debugger looks like:

<img width="622" alt="image" src="https://user-images.githubusercontent.com/49922/188137582-687ece56-fa56-40b5-aafd-c1a20f30eecf.png">

Basically, pdb is squeezed to one line height.

How can I use ipdb / ipdb with Rich live fullscreen applications? Do I need to have some command / hook to release the terminal for pdb?

**Comment by miohtama** [comment]
For the record the trick

```
import os ; os.system("stty sane")
```

does not seem to work.
